scep/authority.go

```diff
@@ -153,14 +151,28 @@ func (a *Authority) GetCACertificates() ([]*x509.Certificate, error) {
 	// Using an RA does not seem to exist in https://tools.ietf.org/html/rfc8894, but is mentioned in
 	// https://tools.ietf.org/id/draft-nourse-scep-21.html. Will continue using the CA directly for now.
 	//
-	// The certificate to use should probably depend on the (configured) Provisioner and may
+	// The certificate to use should probably depend on the (configured) provisioner and may
 	// use a distinct certificate, apart from the intermediate.
 
-	if a.intermediateCertificate == nil {
+	p, err := provisionerFromContext(ctx)
+	if err != nil {
+		return nil, err
+	}
+
+	if len(a.caCerts) == 0 {
 		return nil, errors.New("no intermediate certificate available in SCEP authority")
 	}
 
-	return []*x509.Certificate{a.intermediateCertificate}, nil
+	certs := []*x509.Certificate{}
+	certs = append(certs, a.caCerts[0])
+
+	// NOTE: we're adding the CA roots here, but they are (highly likely) different than what the RFC means.
+	// Clients are responsible to select the right cert(s) to use, though.
+	if p.ShouldIncludeRootsInChain() && len(a.caCerts) > 1 {
+		certs = append(certs, a.caCerts[1:]...)
+	}
```

Comment:
Perhaps we should only add the actual root and not all of them.

Reply:
Changed in https://github.com/smallstep/certificates/pull/797/commits/64680bb16dd076a33e3c1ad53b71b507825c6560. 

I still have my doubts if what we're doing here is in fact what is expected in SCEP, as what they refer to in the RFC as CA seems to be our intermediate CA, but this change shouldn't break things for people already using SCEP.

Reply:
This works in most cases, but it's not strictly right that the first root in `authority.rootX509Certs` is the root that signed the intermediate. But as this special case must be configured in the provisioner, we can force the user to configure the roots accordingly.